common/src/JetCorrections.cxx

```diff
@@ -435,78 +437,98 @@ JetLeptonCleaner_by_KEYmatching::JetLeptonCleaner_by_KEYmatching(uhh2::Context&
 
 JetLeptonCleaner_by_KEYmatching::~JetLeptonCleaner_by_KEYmatching(){}
 
-bool JetLeptonCleaner_by_KEYmatching::process(uhh2::Event& event){
-
-  auto& jets = event.get(h_jets_);
-
-  for(auto& jet : jets){
+bool JetLeptonCleaner_by_KEYmatching::do_cleaning(Jet & jet, uhh2::Event& event) {
+  bool correct_p4(false);
+  auto jet_p4_raw = jet.v4() * jet.JEC_factor_raw();
 
-    bool correct_p4(false);
-    auto jet_p4_raw = jet.v4() * jet.JEC_factor_raw();
-
-    const auto& jet_lepton_keys = jet.lepton_keys();
+  const auto& jet_lepton_keys = jet.lepton_keys();
 
     // muon-cleaning
-    if(event.muons){
-
-      for(const auto& muo : *event.muons){
+  if(event.muons){
 
-        if(muo_id && !(muo_id(muo, event))) continue;
+    for(const auto& muo : *event.muons){
 
-        for(const auto& muo_cand : muo.source_candidates()){
+      if(muo_id && !(muo_id(muo, event))) continue;
 
-          if(std::find(jet_lepton_keys.begin(), jet_lepton_keys.end(), muo_cand.key) != jet_lepton_keys.end()){
+      for(const auto& muo_cand : muo.source_candidates()){
 
-            correct_p4 = true;
+        if(std::find(jet_lepton_keys.begin(), jet_lepton_keys.end(), muo_cand.key) != jet_lepton_keys.end()){
 
-            LorentzVectorXYZE muo_cand_p4;
-            muo_cand_p4.SetPx(muo_cand.px);
-            muo_cand_p4.SetPy(muo_cand.py);
-            muo_cand_p4.SetPz(muo_cand.pz);
-            muo_cand_p4.SetE (muo_cand.E);
+          correct_p4 = true;
 
-            jet_p4_raw -= muo_cand_p4;
-          }
+          LorentzVectorXYZE muo_cand_p4;
+          muo_cand_p4.SetPx(muo_cand.px);
+          muo_cand_p4.SetPy(muo_cand.py);
+          muo_cand_p4.SetPz(muo_cand.pz);
+          muo_cand_p4.SetE (muo_cand.E);
         }
```

Comment:
where is the lepton-candidate p4 removed from jet_p4_raw in this new version? it seems jet_p4_raw is never modified in JetLeptonCleaner_by_KEYmatching::do_cleaning. have you tested the output of these modified routines?